classes/local/store/object_file_system.php

```diff
@@ -585,7 +632,11 @@ public function delete_object_from_hash($contenthash) {
      */
     public function delete_local_file_from_hash($contenthash) {
         $path = $this->get_local_path_from_hash($contenthash);
-        unlink($path);
+        if (unlink($path)) {
+            // Check grandparent dir for empty dirs.
+            $dirpath = str_replace($contenthash, '', $path) . '..';
```

Comment:
this seems awkward, are you just tryng to get the parent dir? I think you want dirname

https://www.php.net/manual/en/function.dirname.php

Reply:
yes, it doesn't work for `5.6` :disappointed: .
-  `7.0.0 | Added the optional levels parameter.`